braid/_braid.c

```diff
@@ -3526,10 +3544,14 @@ _braid_InitHierarchy(braid_Core    core,
       /* We need to ensure an rfactor of 1 for global index 0, and for
        * the case of a user-defined residual function and F-relaxation,
        * a default rfactor value at F-points */
-      rfactors[i] = 1; 
+      rfactors[i] = 1;
    }
    _braid_CoreElt(core, rfactors) = rfactors;
 
+   /* Allocate array of refiment dt values, initialized with NULL */
+   rdtvalues = _braid_CTAlloc(braid_Real*, iupper-ilower+2); /* TODO: +2 needed? */
```

Comment:
From what I can tell, the +2 is probably not needed (?), just ta arrays need the +2, I think.

Reply:
There is a comment a few lines above this for 'rfactor' that says, "ensures non-NULL".  Leaving the +2 here would do the same and maybe that's a good enough reason.  What do you think?  I could simply change the 'TODO' comment.

Reply:
Yeah, you could just remove the TODO comment, so as not to confuse people later.  I honestly don't see why the +2 is needed...but someone did earlier (with the non-NULL comment above), so I'm happy leaving the +2.  It's a trivial amount of memory.